src/poetry/installation/chef.py

```diff
@@ -2,27 +2,206 @@
 
     from poetry.config.config import Config
     from poetry.utils.env import Env
 
 
+class ChefError(Exception):
+    ...
+
+
+class ChefBuildError(ChefError):
+    ...
+
+
+class IsolatedEnv(BaseIsolatedEnv):
+    def __init__(self, env: Env, config: Config) -> None:
+        self._env = env
+        self._config = config
+
+    @property
+    def executable(self) -> str:
+        return str(self._env.python)
+
+    @property
+    def scripts_dir(self) -> str:
+        return str(self._env._bin_dir)
+
+    def install(self, requirements: Collection[str]) -> None:
+        from cleo.io.null_io import NullIO
+        from poetry.core.packages.dependency import Dependency
+        from poetry.core.packages.project_package import ProjectPackage
+
+        from poetry.config.config import Config
+        from poetry.factory import Factory
+        from poetry.installation.installer import Installer
+        from poetry.packages.locker import NullLocker
+        from poetry.repositories.installed_repository import InstalledRepository
+
+        # We build Poetry dependencies from the requirements
+        package = ProjectPackage("__root__", "0.0.0")
+        package.python_versions = ".".join(str(v) for v in self._env.version_info[:3])
+        for requirement in requirements:
+            dependency = Dependency.create_from_pep_508(requirement)
+            package.add_dependency(dependency)
+
+        pool = Factory.create_pool(self._config)
+        installer = Installer(
+            NullIO(),
+            self._env,
+            package,
+            NullLocker(self._env.path.joinpath("poetry.lock"), {}),
+            pool,
+            Config.create(),
+            InstalledRepository.load(self._env),
+        )
+        installer.update(True)
+        installer.run()
+
+
 class Chef:
     def __init__(self, config: Config, env: Env) -> None:
+        self._config = config
         self._env = env
         self._cache_dir = (
             Path(config.get("cache-dir")).expanduser().joinpath("artifacts")
         )
+        self._lock = threading.Lock()
+
+    def prepare(
+        self, archive: Path, output_dir: Path | None = None, *, editable: bool = False
+    ) -> Path:
+        if not self._should_prepare(archive):
+            return archive
+
+        if archive.is_dir():
+            tmp_dir = tempfile.mkdtemp(prefix="poetry-chef-")
+
+            return self._prepare(archive, Path(tmp_dir), editable=editable)
+
+        return self._prepare_sdist(archive, destination=output_dir)
+
+    def _prepare(
+        self, directory: Path, destination: Path, *, editable: bool = False
+    ) -> Path:
+        from subprocess import CalledProcessError
+
+        from poetry.utils.env import EnvManager
+        from poetry.utils.env import VirtualEnv
+
+        with self._lock, temporary_directory() as tmp_dir:
+            EnvManager.build_venv(tmp_dir, executable=self._env.python, with_pip=True)
```

Comment:
We can use `poetry.utils.env.ephemeral_environment` here.

Reply:
In theory, we do not need `pip` nor `setuptools` in there. Also see #5650 for example usage. The major issue with this approach is that it seems that setting up and installing the environments per-package is hefty. However, this change could contribute to this performance no longer being an issue. Additionally, we should cache these environments globally probably keyed with python version and build requires.

Reply:
It would be good if you can pull up the `poetry.utils.pep517` module into this change since that would be reusable across inspection as well.

Reply:
I adapted it to use `ephemeral_environment`. Pulling `poetry.utils.pep517` in is a bit too much for me (at least for now). This should probably not block this PR and we can refactor it later in a follow-up PR.